pvfactors/irradiance/models.py

```diff
@@ -240,6 +293,17 @@ def fit(self, timestamps, DNI, DHI, solar_zenith, solar_azimuth,
                 timestamps, DNI, DHI, solar_zenith, solar_azimuth,
                 surface_tilt, surface_azimuth)
 
+        # Save and calculate total POA values from Perez model
+        if GHI is None:
+            # Calculate GHI if not specified
+            GHI = DNI * cosd(solar_zenith) + DHI
+        self.GHI = GHI
+        self.DHI = DHI
+        perez_front_pvrow = get_total_irradiance(
+            surface_tilt, surface_azimuth, solar_zenith, solar_azimuth,
+            DNI, GHI, DHI, albedo=albedo)
+        total_perez_front_pvrow = perez_front_pvrow['poa_global']
```

Comment:
It looks like there are a lot of common things with `IsotropicOrdered`, maybe it should be regrouped / sub-classed?

Reply:
Very good point, I could definitely regroup the classes here and it's not fun updating things separately like that. 
But the reason why I've done it this way is that I am really trying to emphasize to the users that they can create completely separate irradiance models by creating their own classes, and still be able to use it in pvfactors... that's why I've worked on abstracting things so much in pvfactors, so that it can be used like sort of a framework as well